text/0000-result-expect.md

```diff
@@ -0,0 +1,40 @@
+- Feature Name: `result_expect`
+- Start Date: 2015-05-13
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add an `expect` method to the Result type, bounded to `E: Debug`
```

Comment:
`unwrap` and `expect` are very similar methods. `expect` is intended to also to not be for user consumption (it panics after all).
The intent is to provide a version of `unwrap` that provides context on the error.